### PR TITLE
Followers feature

### PR DESCRIPTION
**Pull request by janegreene**
Added ability to see who a user follows and who is following them. 
Made PORO's for Follower and Following.
Updated views to display results from API.

### CODE REVIEW

File: app/models/following.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Following
```
Comment (janegreene) [inline app/models/following.rb:1]: Potential for refactor as inheritance.

File: spec/features/user/user_can_see_github_followers_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  it 'can see github followers' do
+    user_2 = create(:user)
+    user_2.update(github_token: ENV["GITHUB_TOKEN_A"])
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user_2)
+    visit dashboard_path
+
+    expect(page).to have_content("Github")
+    within ".github" do
+      expect(page).to have_content("Followers")
+      expect(page).to have_link(ENV["GITHUB_USERNAME_C"])
```
Comment (janegreene) [inline spec/features/user/user_can_see_github_followers_spec.rb:13]: Made to variable to protect the innocent.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -5,11 +5,21 @@ def show
...
+    @parsed_repo_resp.map! do |resp|
       Repo.new(resp)
     end
+    followers_resp = conn.get('user/followers')
+    @parsed_follow_resp = JSON.parse(followers_resp.body, symbolize_names: true)
+    @parsed_follow_resp.map! do |follower|
+      Follower.new(follower)
+    end
+    following_resp = conn.get('user/following')
+    @parsed_flwing_resp = JSON.parse(following_resp.body, symbolize_names: true)
```
Comment (arpariseau) [inline app/controllers/users_controller.rb:19]: see if we can figure out a way to keep it below 80 characters but make it more clear than "flwing"

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -7,11 +7,21 @@
     <li> <%= current_user.first_name  %> <%= current_user.last_name %> </li>
     <li> <%= current_user.email%> </li>
   </ul>
-  <% if !@parsed_resp.nil? %>
-    <section><h3>Github</h3>
-    <% @parsed_resp.each do |resp| %>
+  <% if !@parsed_repo_resp.nil? %>
+    <section class="github"><h3>Github</h3>
```
Comment (arpariseau) [inline app/views/users/show.html.erb:11]: good to have this as a class so we can look within it

**Review by arpariseau — APPROVED** [review]

**Pull request merged**